flang/lib/Optimizer/Transforms/ExternalNameConversion.cpp

```diff
@@ -101,9 +108,11 @@ struct MangleNameOnAddrOfOp : public mlir::OpRewritePattern<fir::AddrOfOp> {
                   mlir::PatternRewriter &rewriter) const override {
     auto result = fir::NameUniquer::deconstruct(op.symbol().getRootReference());
     if (fir::NameUniquer::isExternalFacingUniquedName(result)) {
-      auto newName = rewriter.getSymbolRefAttr(mangleExternalName(result));
+      auto newName = mangleExternalName(result);
+      auto newNameAttr = rewriter.getSymbolRefAttr(mangleExternalName(result));
       rewriter.replaceOpWithNewOp<fir::AddrOfOp>(op, op.resTy().getType(),
-                                                 newName);
+                                                 newNameAttr);
+      SymbolTable::setSymbolName(op, newName);
```

Comment:
I am not familiar at all with how the MLIR symbol table works, but is it OK to call this after `op` was replaced ?

In general though, do you really need to/is it correct to call `SymbolTable::setSymbolName(op, newName)` on other ops than fir::GlobalOp and mlir::FuncOp ? It seems they are the only ones with the `Symbol` interface in FIROps.td, and all the other ones are using a name, but not really defining it, so I am surprised it is also required to update the symbol table for those.

Reply:
Good point. Let's keep it only for `GlobalOp` and `FuncOp`. My guess is that you can call it for whatever but it might pollute the symbol table with stuffs you don't really want.